packages/react-native/Libraries/Modal/Modal.js

```diff
@@ -173,53 +154,35 @@ function confirmProps(props: Props) {
   }
 }
 
-class Modal extends React.Component<Props> {
+class Modal extends React.Component<Props, State> {
   static defaultProps: {|hardwareAccelerated: boolean, visible: boolean|} = {
     visible: true,
     hardwareAccelerated: false,
   };
 
   static contextType: React.Context<RootTag> = RootTagContext;
 
-  _identifier: number;
-  _eventSubscription: ?EventSubscription;
-
   constructor(props: Props) {
     super(props);
+    this.state = {
+      isRendering: props.visible === true,
+    };
     if (__DEV__) {
       confirmProps(props);
     }
-    this._identifier = uniqueModalIdentifier++;
   }
 
-  componentDidMount() {
-    // 'modalDismissed' is for the old renderer in iOS only
-    if (ModalEventEmitter) {
-      this._eventSubscription = ModalEventEmitter.addListener(
-        'modalDismissed',
-        event => {
-          if (event.modalID === this._identifier && this.props.onDismiss) {
-            this.props.onDismiss();
-          }
-        },
-      );
-    }
-  }
-
-  componentWillUnmount() {
-    if (this._eventSubscription) {
-      this._eventSubscription.remove();
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.visible !== true && this.props.visible === true) {
```

Comment:
What is the purpose of `isRendering?

isRendering is mapped to value of `visible` prop.

Reply:
`isRendering` is being used to keep the Modal rendered until the `onDismiss` callback is called.

Reply:
What if the parent component unmounts Modal?

Reply:
In that case, `onDismiss` won't be called.